### ISSUE TITLE
Timeout issues on some SDK calls

### ISSUE THREAD

**Issue by mickaelpois** [Mangopay/mangopay2-php-sdk#297]
Some PHP SDK users are always complaining about random timeouts when calling our services.

Most of time, calls has been successfully received by our servers and Payins, Transferts, Payouts, etc. are well processed.

We will try in the following months to find some ways to enhance this, avoiding extra manual checks to ensure call has been done safely.

**Comment by iammichiel** [comment]
Hi,  

I think I was one of the main reporters on this issue. This still occurs for us. Is there any progress on this? Is this happening only using the php-sdk? 

I am ready to use our environment for additional debugging if required.

**Comment by SoloJr** [comment]
@mickaelpois I believe this is not an issue anymore. Can you take a look?

**Comment by mangomaxoasis** [comment]
Hi @iammichiel,
Indeed in certain very isolated periods the API responses (this is independent of the SDK) have experienced slowdowns, we actively monitor and inform our user base of issues of this kind.
At the moment the responses speeds are fully operational and compliant.
For info, the timeout periods in this SDK are of 30secs for connection, and 80s for responses.
Cheers,
Max

**Comment by iammichiel** [comment]
Just for you info, we experienced another timeout last week : 

More precisely : `Jun 11, 2020 7:27:03 AM UTC`
When calling : 
```php
$mangoUser = $this->mango->Users->Get($user->getMangoId());
```

Might not want to consider this as closed.

**Comment by mangomaxoasis** [comment]
Hi @iammichiel,
I'll investigate this, to confirm it was one occurence at  Jun 11, 2020 7:27:03 AM UTC ?

**Comment by iammichiel** [comment]
Yes, we had only one occurence of the timeout over the last 7 days. THe logs do not go further. I will let you know if this occurs again.

**Comment by mangomaxoasis** [comment]
Thanks @iammichiel !

**Comment by iammichiel** [comment]
We just experienced another timeout : 

` $cardRegistration = $mangoPayApi->CardRegistrations->Get($cardId);`

> Jun 30, 2020 9:28:13 AM UTC

**Comment by mangomaxoasis** [comment]
Hi @iammichiel we're looking into it

**Comment by catacraciun** [comment]
@iammichiel is his happening on staging, or production?

**Comment by mangomaxoasis** [comment]
@iammichiel also, can you confirm the version of the SDK?

**Comment by iammichiel** [comment]
I wouldn't bother you if this happened on staging. :) This is a **production** issue. 

We are using the 3.0.0 version of the SDK, but this has been the case for a couple of months, meaning, it also occurred using the 2.x version of the SDK. 

If this has any value for you, our env making those calls is hosted on AWS in the eu-west-region (Ireland).

**Comment by iammichiel** [comment]
And another timeout : Jul 4, 2020 9:02:00 AM UTC.

**Comment by iammichiel** [comment]
And another timeout this morning : `Jul 22, 2020 7:09:37 AM UTC`

**Comment by mangomaxoasis** [comment]
Hi @iammichiel 
We are still having trouble isolating this issue, would you be free for a quick call so we can better understand the situation?

**Comment by iammichiel** [comment]
Yeah, sure. Perhaps, we can switch to a email to setup that call ? You can contact me at michiel at ouistock dot fr.

**Comment by mangomaxoasis** [comment]
Hi @iammichiel mail sent, let me know if you don't receive it 
Max

**Comment by spiritinlife** [comment]
Happy to see this has been reported by others. We are also experiencing the same from time to time. It can be sporadic with just one timeout or can also last for multiple requests 10-20 feeling like an outage.

I have been in talks with the support to provide specific api calls that failed and will revert if we get any more information. It seems that mangopay responds with 200 but not under 30 seconds.
